src/entity/units/goru.ts

```diff
@@ -44,50 +60,328 @@ const unit: UnitSource = {
     death: 'goruDeath'
   },
   init: (unit: Unit.IUnit, underworld: Underworld) => {
+    // TODO - Bug: Undying 2 can be re-added by cloning/splitting/creating a Goru with no undying modifier
+    // Using originalLife might prevent this, but also prevents summoned Goru's from getting undying.
+    if (!unit.modifiers[undyingModifierId]) {
+      Unit.addModifier(unit, undyingModifierId, underworld, false, 2);
```

Comment:
I think we need to add some floating text to explain what's resurrecting the unit when they rez via undying.

Reply:
Just an idea: could undying be a curse so it could be purified? That'd be a cool synergy.  Would that wreck anything?

Reply:
Undying feels more like a blessing to me, but I could see it working as a curse too I guess. I don't think I like this interaction though. It would kind've trivialize a cool fight that we want people to see, I think. Though, it's not too different from bone shrapnel in that sense. Might be a little crazy with Contaminate. I don't think it would *break* anything super bad

Reply:
Floating text definitely a good idea!

Reply:
Okay that makes sense, let's leave it as a blessing then